packages/premint-sdk/src/premint-api.ts

```diff
@@ -0,0 +1,684 @@
+import { createPublicClient, decodeEventLog, http } from "viem";
+import type {
+  Account,
+  Address,
+  Chain,
+  Hex,
+  PublicClient,
+  TransactionReceipt,
+  WalletClient,
+} from "viem";
+import {
+  zoraCreator1155PremintExecutorImplABI,
+  zoraCreator1155PremintExecutorImplAddress,
+  zoraCreatorFixedPriceSaleStrategyAddress,
+} from "@zoralabs/zora-1155-contracts";
+import { foundry, zora, zoraTestnet } from "viem/chains";
+import { PremintConfig, preminterTypedDataDefinition } from "./preminter";
+
+export type NetworkConfig = {
+  chainId: number;
+  zoraPathChainName: string;
+  zoraBackendChainName: string;
+  isTestnet: boolean;
+};
+
+export const enum BackendChainNames {
+  ZORA_MAINNET = "ZORA-MAINNET",
+  ZORA_GOERLI = "ZORA-GOERLI",
+}
+
+async function wait(delayMs: number) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, delayMs);
+  });
+}
+
+const ZORA_API_BASE = "https://api.zora.co/premint/";
+
+export const networkConfigByChain: Record<number, NetworkConfig> = {
+  [zora.id]: {
+    chainId: zora.id,
+    isTestnet: false,
+    zoraPathChainName: "zora",
+    zoraBackendChainName: BackendChainNames.ZORA_MAINNET,
+  },
+  [zoraTestnet.id]: {
+    chainId: zora.id,
+    isTestnet: true,
+    zoraPathChainName: "zgor",
+    zoraBackendChainName: BackendChainNames.ZORA_GOERLI,
+  },
+  [foundry.id]: {
+    chainId: foundry.id,
+    isTestnet: true,
+    zoraPathChainName: "zgor",
+    zoraBackendChainName: BackendChainNames.ZORA_GOERLI,
+  },
+};
+
+export class BadResponse extends Error {
+  status: number;
+  json: any;
+  constructor(message: string, status: number, json: any) {
+    super(message);
+    this.name = "BadResponse";
+    this.status = status;
+    this.json = json;
+  }
+}
+
+type MintArgumentsSettings = {
+  tokenURI: string;
+  maxSupply?: bigint;
+  maxTokensPerAddress?: bigint;
+  pricePerToken?: bigint;
+  mintStart?: bigint;
+  mintDuration?: bigint;
+  royaltyMintSchedule?: number;
+  royaltyBPS?: number;
+  royaltyRecipient?: Address;
+  fixedPriceMinter?: Address;
+};
+
+const OPEN_EDITION_MINT_SIZE = "18446744073709551615";
+export const DefaultMintArguments = {
+  maxSupply: BigInt(OPEN_EDITION_MINT_SIZE),
+  maxTokensPerAddress: 0n,
+  pricePerToken: 0n,
+  mintDuration: BigInt(60 * 60 * 24 * 7), // 1 week
+  mintStart: 0n,
+  royaltyMintSchedule: 0,
+  royaltyBPS: 1000, // 10%,
+};
+
+function getLogFromReceipt(receipt: TransactionReceipt) {
```

Comment:
really what an end account would need from this is to get the token id from the logs. so how about just simplifying this to get the `tokenId`, renaming it to:

`getTokenIdFromLogs`.

also I'd export this function so people can call it directly in the case they are not using the api class

Reply:
What about contract address as well?

renamed to `getPremintedLogFromReceipt`